invenio_drafts_resources/services/records/components/files.py

```diff
@@ -36,10 +36,10 @@ def create(self, identity, data=None, record=None, errors=None):
               (this interface is used in records-resources and rdm-records)
         """
         draft = record
-        if self.service.check_permission(identity, "manage_files"):
-            enabled = data.get("files", {}).get("enabled", True)
-            draft.files.enabled = enabled
-        else:
+        enabled = data.get("files", {}).get("enabled", True)
+
+        can_manage_files = self.service.check_permission(identity, "manage_files")
+        if draft.files.enabled != enabled and not can_manage_files:
```

Comment:
Should the first check be done before the permission check to save db queries in case is not changed?

(pseudocode)

```suggestion
        if draft.files.enabled != enabled:
            if self.service.check_permission(identity, "manage_files"):
                ....
```

or nest both in the same line, readibility in this case (i.e. instantiate `can_manage_files`) involves many db/es queries 😓